crates/driver/src/tests/cases/negative_scores.rs

```diff
@@ -23,12 +23,32 @@ async fn no_valid_solutions() {
 
 #[tokio::test]
 #[ignore]
-async fn one_valid_solution() {
+async fn one_valid_solution_with_solver_score() {
     let test = setup()
         .pool(ab_pool())
         .order(ab_order().rename("no surplus").no_surplus())
         .order(ab_order())
-        .solution(ab_solution())
+        .solution(ab_solution().score(Score::Solver(DEFAULT_SCORE_MIN.into())))
```

Comment:
Can we add a test here that solutions with different scores/success probabilities get ranked accordingly (can be a single test case)?

Reply:
Added in a separate test file: https://github.com/cowprotocol/services/pull/1856/commits/fd677479a68e53eb9b782a16bb30cde0e7365726#diff-a192474ce5a41f10c12020b8de582f0726597fd43665449af9f9f9252cde652b